build/pkgs/sphinx/distros/conda.txt

```diff
@@ -1 +1 @@
-sphinx<6,>=5.2
```

Comment:
why? I would only add a version constraint if there are really known issues.

Reply:
Because we know that in every Sphinx upgrade we had to fix something critical.

Reply:
That probably applies to many dependencies, but almost none of them have upper constraints in the conda info.

Reply:
No, I am saying that it specifically applies to Sphinx.

Reply:
The description does not say you will allow sphinx greater than 7, just allow 6 and 7. These days some other packages break doctests that were too "precise" but very few break the build like sphinx do with documentation. We carry a lot of sphinx baggage from some early day of sage, and almost each sphinx major release is a reminder that we need to do something about it.

Reply:
Exactly; in particular, #30893 (autodoc) is still unresolved.

Reply:
> issues from random dependency updates

Once again, Tobias, this concern is specifically about Sphinx.

Reply:
> What's the problem with adding the version constraint once v8 is released and actually breaks things

The problem is that that's too late.

Reply:
why is this too late? The conda files are only used for devs and they can easily override it. Note also that until a few months ago the conda files for sphinx had no version constraints...

Reply:
And how is this specific to sphinx? From what I've seen most new versions of say scipy also result in failing doctests.

Reply:
> we should use conda-lock but not some preemptive constraints

False dichotomy. As you know, one can do both, as they serve separate purposes.

Reply:
> And how is this specific to sphinx?

@kiwifb explained it above.

Reply:
> The conda files are only used for devs and they can easily override it.

No, we cannot assume that the people who follow our instructions for installing Sage are already familiar with these conda-specific details.

Reply:
> > The conda files are only used for devs and they can easily override it.
> 
> No, we cannot assume that the people who follow our instructions for installing Sage are already familiar with these conda-specific details.

It would only be for the one week between the discovery that v8 breaks things in a serious way and merging the PR that adds the upper constraint. Also conda is marked experimental, so a few rough edges are acceptable...finally, its not conda specific. Its quite common that one has to add additional upper version constraints for dependencies when using `pip install `.

And I still don't see how sphinx is special. Sure, experience says that v8 will likely fail and sage's doc generation is quite tightly bound to sphinx. But all of this applies to cython as well in an even more extreme way, and there we don't specify any upper bound.

Reply:
> finally, its not conda specific.

We agree here -- I of course also object to removing the upper bound from `build/pkgs/sphinx/install-requires.txt`.

Reply:
> conda is marked experimental, so a few rough edges are acceptable.

That's a terrible argument; certainly we don't want to keep it "experimental".

Reply:
Thanks for your input, I see your point but don't agree with it.

Reply:
Which point? I've made many.